common/djangoapps/course_modes/views.py

```diff
@@ -139,14 +139,11 @@ def get(self, request, course_id, error=None):  # lint-amnesty, pylint: disable=
         # If there isn't a verified mode available, then there's nothing
         # to do on this page.  Send the user to the dashboard.
         if not CourseMode.has_verified_mode(modes):
-            return redirect(reverse('dashboard'))
+            return self._redirect_to_course_or_dashboard(course, course_key, request.user)
```

Comment:
Note: the code for redirecting the learner if there was only a verified mode was already partially here, so I chose to continue it here rather than move it to the bottom.